node/src/Pos/Explorer/Txp/Local.hs

```diff
@@ -126,7 +129,10 @@ eTxProcessTransactionNoLock itw@(txId, txAux) = runExceptT $ do
             M.fromList $
             catMaybes $
             toList $ NE.zipWith (liftM2 (,) . Just) _txInputs resolvedOuts
-    curTime <- currentTimeSlotting
+    -- First get the current @SlotId@ so we can calculate the time.
+    -- Then get when that @SlotId@ started and use that as a time for @Tx@.
+    mTxTimestamp <- getCurrentSlotBlocking >>= getSlotStart
```

Comment:
You can avoid using `getCurrentSlotBlocking` here. This function shouldn't block if current slot is unknown. The reason it will work is that there is
```
    epoch <- siEpoch <$> (note ToilSlotUnknown =<< getCurrentSlot)
```
above. So if we don't know current slot, it will finish earlier. I would recommend removing this line (which makes impression that this line can block) and write:
```
  slot <- (note ToilSlotUnknown =<< getCurrentSlot)
  let epoch = siEpoch slot
  mTxTimestamp <- getSlotStart slot
```
It can also be slightly more efficient.

Reply:
Ok, will fix, thanks.